core/src/Kokkos_HIP_Space.hpp

```diff
@@ -687,6 +687,7 @@ class HIP {
   };
 
   int hip_device() const;
+  static hipDeviceProp_t const& hip_device_prop();
```

Comment:
Note that this function is not declared static in the CUDA backend.

Reply:
Doesn't this, in general, depend on the device the `ExecutionSpace` was initialized with? What if there are multiple GPUs with different properties?

Reply:
One is selected when Kokkos is initialized and that's it.

Reply:
OK. `HIP::impl_initialize` creates the singleton and then calls `HIPIntenal::initialize()`. In that function, we use a `HIPInternalDevices` singleton that stores the information for all the devices and then set the properties for the device requested in `HIP::impl_initialize`.